### ISSUE TITLE
Redirecting to endpoint with data

### ISSUE THREAD

**Issue by Ahtii** [tiangolo/fastapi#2288]
Hi,

I have two endpoint `/api/register` and `/api/sociallogin`.

**register endpoint:**

```
@app.post("/api/register/")
async def register(user: validators.RegisterValidator, db: Session = Depends(get_db)):
    return views.register(db, user)
```

**social login endpoint:**

```
@app.post("/api/sociallogin", include_in_schema=False)
def social_login(request: Request, data: validators.SocialLoginValidator):
    data = views.social_login(request, data)
    response = RedirectResponse(url="/api/register", data)   # how to redirect to regsiter endpoint with user data ?
    return response
```

I want to redirect to register endpoint from sociallogin endpoint with the user data how would I do it ?

**Comment by Mause** [comment]
Have you tried using query parameters?

**Comment by Ahtii** [comment]
@Mause No can i use it without having to change the register function parameters ?

**Comment by Mause** [comment]
You are passing new data, you have to change something, no?

**Comment by Mause** [comment]
You could also store it in a cookie, or in a database - your requirements are very vague

**Comment by Ahtii** [comment]
I am just trying to register/save the user after authenticating him/her from a social provider. I am reusing the same endpoint how is that vague ? can you please elaborate which part makes it vague ?

**Comment by Ahtii** [comment]
@Mause yes I am passing a new data which I want to replace with first parameter of register endpoint so that it would be validated and

**Comment by tiangolo** [comment]
Thanks for the help @Mause! 🍰 

I imagine you solved your need, thanks for coming back to close the issue @Ahtii ☕ 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
